src/test/java/com/example/produktapi/service/ExceptionHandlerAdviceTest.java

```diff
@@ -0,0 +1,67 @@
+package com.example.produktapi.service;
+
+import com.example.produktapi.exception.BadRequestException;
+import com.example.produktapi.exception.EntityNotFoundException;
+import com.example.produktapi.model.Product;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ExceptionHandlerAdviceTest {
+
+    private ExceptionHandlerAdvice exceptionHandlerAdvice;
+
+    @BeforeEach
+    public void setup() {
+        exceptionHandlerAdvice = new ExceptionHandlerAdvice();
+    }
+
+    @Test
+    @DisplayName("unitTestofHandleEntityNotFoundException")
+    void handleEntityNotFoundException() {
+        //Arrange
+        EntityNotFoundException entityNotFoundException = new EntityNotFoundException(0);
+        //Act
+        ResponseEntity<String> response = exceptionHandlerAdvice.handleException(entityNotFoundException);
+        //Assert
+        assertAll(
+                () ->assertEquals(HttpStatus.NOT_FOUND, response.getStatusCode()),
+                () ->assertEquals("Produkt med id 0 hittades inte", response.getBody())
+        );
+    }
+
+    @Test
+    @DisplayName("unitTestofHandleBadRequestException")
+    void handleBadRequestException() {
+        //Arrange
+        String title = "Example Product";
+        Product product = new Product();
+        product.setTitle(title);
+        BadRequestException badRequestException = new BadRequestException("En produkt med titeln: "+ product.getTitle() + " finns redan");
```

Comment:
Här testar du i onödan även klassen `Product`, således är detta ej ett enhetstest. Lösningen på detta är att ta bort användningen av `Product` och inkludera titeln direkt i felmeddelandet.

Se nedan förslag på lösning.

```suggestion
        BadRequestException badRequestException = new BadRequestException("En produkt med titeln: Example Product finns redan");
```

Reply:
Här snurrade jag till det ordentligt. Ditt förslag gör testet mycket mer avskalat vilket är bra. Men jag förstår inte vad det är som testas. Jag uppfattar ditt förslag som cirkulärt och det var här jag gick vilse första gången. Jag fick för mig att man ville testa att textsträngen som hämtas från exceptionsklasserna stämmer med förväntat utfall. Jag frågar dig mer vid nästa sprintmöte.